qiskit/algorithms/state_fidelities/base_state_fidelity.py

```diff
@@ -0,0 +1,305 @@
+    def _preprocess_values(
+        circuits: QuantumCircuit | Sequence[QuantumCircuit],
+        values: Sequence[float] | Sequence[Sequence[float]] | None = None,
+    ) -> Sequence[Sequence[float]]:
+        """
+        Checks whether the passed values match the shape of the parameters
+        of the corresponding circuits and formats values to 2D list.
+
+        Args:
+            circuits: List of circuits to be checked.
+            values: Parameter values corresponding to the circuits to be checked.
+
+        Returns:
+            A 2D value list if the values match the circuits, or an empty 2D list
+            if values is None.
+
+        Raises:
+            ValueError: if the number of parameter values doesn't match the number of
+                        circuit parameters
+            TypeError: if the input values are not a sequence.
+        """
+
+        if isinstance(circuits, QuantumCircuit):
+            circuits = [circuits]
+
+        if values is None:
+            for circuit in circuits:
+                if circuit.num_parameters != 0:
+                    raise ValueError(
+                        f"`values` cannot be `None` because circuit <{circuit.name}> has "
+                        f"{circuit.num_parameters} free parameters."
+                    )
+            return [[]]
+        else:
+
+            # Support ndarray
+            if isinstance(values, np.ndarray):
+                values = values.tolist()
+            if len(values) > 0 and isinstance(values[0], np.ndarray):
+                values = [v.tolist() for v in values]
+
+            if not isinstance(values, Sequence):
+                raise TypeError(
+                    f"Expected a sequence of numerical parameter values, "
+                    f"but got input type {type(values)} instead."
+                )
+
+            # ensure 2d
+            if len(values) > 0 and not isinstance(values[0], Sequence):
+                values = [values]
+            return values
+
+    def _check_qubits_match(self, circuit_1: QuantumCircuit, circuit_2: QuantumCircuit) -> None:
+        """
+        Checks that the number of qubits of 2 circuits matches.
+        Args:
+            circuit_1: (Parametrized) quantum circuit.
+            circuit_2: (Parametrized) quantum circuit.
+
+        Raises:
+            ValueError: when ``circuit_1`` and ``circuit_2`` don't have the
+            same number of qubits.
+        """
+
+        if circuit_1.num_qubits != circuit_2.num_qubits:
+            raise ValueError(
+                f"The number of qubits for the first circuit ({circuit_1.num_qubits}) "
+                f"and second circuit ({circuit_2.num_qubits}) are not the same."
+            )
+
+    @abstractmethod
+    def create_fidelity_circuit(
+        self, circuit_1: QuantumCircuit, circuit_2: QuantumCircuit
+    ) -> QuantumCircuit:
+        """
+        Implementation-dependent method to create a fidelity circuit
+        from 2 circuit inputs.
```

Comment:
```suggestion
        from 2 circuit inputs.

```
Needs a blank line otherwise the Args get rolled into the text

![image](https://user-images.githubusercontent.com/40241007/188219183-0e42a8c9-a609-43fe-970f-d78a950e1d0b.png)